pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,413 @@
+	Openshift                            bool
+	PolicyRecommendationServerCertSecret certificatemanagement.KeyPairInterface
+	PullSecrets                          []*corev1.Secret
+	TrustedBundle                        certificatemanagement.TrustedBundle
+
+	// Whether or not the cluster supports pod security policies.
+	UsePSP bool
+}
+
+type policyRecommendationComponent struct {
+	cfg   *PolicyRecommendationConfiguration
+	image string
+}
+
+func PolicyRecommendation(cfg *PolicyRecommendationConfiguration) Component {
+	return &policyRecommendationComponent{
+		cfg: cfg,
+	}
+}
+
+func (pr *policyRecommendationComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := pr.cfg.Installation.Registry
+	path := pr.cfg.Installation.ImagePath
+	prefix := pr.cfg.Installation.ImagePrefix
+
+	var err error
+	pr.image, err = components.GetReference(components.ComponentPolicyRecommendation, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (pr *policyRecommendationComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeAny
+}
+
+func (pr *policyRecommendationComponent) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		// In order to switch to a restricted namespace, we need to set:
+		// - securityContext.capabilities.drop=["ALL"]
+		// - securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost"
+		CreateNamespace(PolicyRecommendationNamespace, pr.cfg.Installation.KubernetesProvider, PSSBaseline),
+		allowTigeraPolicyForPolicyRecommendation(pr.cfg),
+	}
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.PullSecrets...)...)...)
+
+	objs = append(objs,
+		pr.serviceAccount(),
+		pr.clusterRole(),
+		pr.clusterRoleBinding(),
+		pr.deployment(),
+	)
+
+	if pr.cfg.KeyValidatorConfig != nil {
+		objs = append(objs, secret.ToRuntimeObjects(pr.cfg.KeyValidatorConfig.RequiredSecrets(PolicyRecommendationNamespace)...)...)
+		objs = append(objs, configmap.ToRuntimeObjects(pr.cfg.KeyValidatorConfig.RequiredConfigMaps(PolicyRecommendationNamespace)...)...)
+	}
+
+	if pr.cfg.TrustedBundle != nil {
+		objs = append(objs, pr.cfg.TrustedBundle.ConfigMap(PolicyRecommendationNamespace))
+	}
+	return objs, nil
+}
+
+func (pr *policyRecommendationComponent) Ready() bool {
+	return true
+}
+
+func (pr *policyRecommendationComponent) clusterRole() client.Object {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{"licensekeys", "managedclusters"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"crd.projectcalico.org"},
+			Resources: []string{"licensekeys"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"crd.projectcalico.org", "projectcalico.org"},
```

Comment:
same thing with the crd.projectcalico.org in this group - the controller should jsut be reading the CRs